### ISSUE TITLE
Message#awaitMessageComponent is not responding

### ISSUE THREAD

**Issue by Raiden-Ei** [discordjs/discord.js#7336]
### Which package is this bug report for?

discord.js

### Issue description

1. Install latest dev version of Discord.js
2. Use `Message#awaitMessageComponent` function

Expected behavior: the function returns `Promise<MessageComponentInteraction>`
Actual behavior: the function returns nothing.

### Code sample

```typescript
const { ActionRow, ButtonComponent } = require('discord.js');

const actionRow = new ActionRow()
  .addComponents(
    new ButtonComponent()
      .setStyle(2)
      .setCustomId('TEST')
      .setLabel('TEST')
  );

const message = await interaction.editReply({ content: 'hi', components: [actionRow] });
const component = await message.awaitMessageComponent({
  filter: (i) => i.user.id === interaction.user.id,
  time: 30_000,
});
console.log(component);
```

### Package version

14.0.0-dev (d2d3a80c556a104099a1ddb1b24f1b921c553257)

### Node.js version

17.4.0

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

CHANNEL

### Which gateway intents are you subscribing to?

GUILDS, GUILD_MEMBERS, GUILD_PRESENCES, GUILD_MESSAGES

### I have tested this issue on a development release

d2d3a80c556a104099a1ddb1b24f1b921c553257

**Comment by suneettipirneni** [comment]
I think I may know what's happening here, I'll try to submit a fix soon.

**Comment by Raiden-Ei** [comment]
> I think I may know what's happening here, I'll try to submit a fix soon.

Sooo.. what's going on? did you find the problem?

**Comment by suneettipirneni** [comment]
@Raiden-Ei PR to fix is now up #7341
